Add options to enable creating users as privileged, and to

From:  <>

enable users being merged.  Done via new official config 
variables, so these flags can be turned on in config files too.
---

 bin/rtldapimport               |   11 +++++++++--
 lib/RT/Extension/LDAPImport.pm |    3 ++-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/bin/rtldapimport b/bin/rtldapimport
index 09900d2..efaf9f4 100644
--- a/bin/rtldapimport
+++ b/bin/rtldapimport
@@ -11,9 +11,11 @@ RT::Init;
 
 use RT::Extension::LDAPImport;
 
-my ($debug, $help, $import, @ldap_files, $nositeconfig );
+my ($debug, $help, $import, @ldap_files, $nositeconfig, $modexist, $privileged );
 use Getopt::Long;
-GetOptions ( "debug" => \$debug, "import" => \$import, "ldapdir=s" => \@ldap_files,"nositeconfig" => \$nositeconfig, "help" => \$help );
+GetOptions ( "debug" => \$debug, "import" => \$import, "ldapdir=s" => \@ldap_files,
+             "nositeconfig" => \$nositeconfig, "help" => \$help,
+             "modexisting" => \$modexist, "privileged" => \$privileged );
 if ($help) {  
    print "$0: [--debug] [--import] [--ldapdirs=file] [--help]\n";
    print "   --help           This usage statement.\n";
@@ -24,10 +26,15 @@ if ($help) {
    print "   --nositeconfig   If set, no imports will be performed with default values in";
    print "                    RT_SiteConfig.pm.  Only settings from files specified \n";
    print "                    with --ldapdirs will be consulted\n";
+   print "   --modexisting    Update identities in RT that already exist.\n";
+   print "   --privileged     Turn on privileged flag.  Use with --modexisting to\n";
+   print "                    make previously imported unprivileged accounts privileged.\n";
    print "   --debug          Enable debugging.\n";
    exit 0;
 }   
 
+$RT::LDAPImportPrivileged = 1 if $privileged; 
+$RT::LDAPUpdateUsers = 1 if $modexisting; 
 
 my $importer = RT::Extension::LDAPImport->new;
 $importer->screendebug(1) if $debug;
diff --git a/lib/RT/Extension/LDAPImport.pm b/lib/RT/Extension/LDAPImport.pm
index 009cde0..58087f6 100644
--- a/lib/RT/Extension/LDAPImport.pm
+++ b/lib/RT/Extension/LDAPImport.pm
@@ -319,13 +319,14 @@ sub create_rt_user {
         my $message = "User $user->{Name} already exists as ".$user_obj->Id;
         if ($RT::LDAPUpdateUsers) {
             $self->_debug("$message, updating their data");
+	    $user->{Privileged} = 1 if $RT::LDAPImportPrivileged;
             my @results = $user_obj->Update( ARGSRef => $user, AttributesRef => [keys %$user] );
             $self->_debug(join(':',@results||'no change'));
         } else {
             $self->_debug("$message, skipping");
         }
     } else {
-        my ($val, $msg) = $user_obj->Create( %$user, Privileged => 0 );
+        my ($val, $msg) = $user_obj->Create( %$user, Privileged => $RT::LDAPImportPrivileged );
 
         unless ($val) {
             $self->_error("couldn't create user_obj for $user->{Name}: $msg");
